esp-wifi/src/wifi/mod.rs

```diff
@@ -1541,13 +1541,15 @@ unsafe extern "C" fn recv_cb_sta(
             queue.push_back(packet);
             true
         } else {
+            mem::forget(packet);
```

Comment:
Please add a comment that we're not allowed to drop `packet` here, otherwise we'll call `esp_wifi_internal_free_rx_buffer` in a critical section which is apparently NotGood(TM)

Reply:
you already added that comment loooong time ago a few lines above

Reply:
really long time ago:

![image](https://github.com/user-attachments/assets/6dabe174-a270-421b-9507-08a22478249f)

Reply:
Huh. Okay. What would you say to a different solution, that doesn't involve `forget`? We could return `Ok(()) / Err(packet)`, and dropping that would free the buffer.

Reply:
sure that should work and probably looks a little bit better